src-java/kilda-persistence-tinkerpop/src/main/java/org/openkilda/persistence/ferma/frames/HaFlowFrame.java

```diff
@@ -0,0 +1,373 @@
+    @Convert(FlowStatusConverter.class)
+    public abstract FlowStatus getStatus();
+
+    @Override
+    @Property(STATUS_PROPERTY)
+    @Convert(FlowStatusConverter.class)
+    public abstract void setStatus(FlowStatus status);
+
+    @Override
+    @Property(MAX_LATENCY_PROPERTY)
+    public abstract Long getMaxLatency();
+
+    @Override
+    @Property(MAX_LATENCY_PROPERTY)
+    public abstract void setMaxLatency(Long maxLatency);
+
+    @Override
+    @Property(MAX_LATENCY_TIER_2_PROPERTY)
+    public abstract Long getMaxLatencyTier2();
+
+    @Override
+    @Property(MAX_LATENCY_TIER_2_PROPERTY)
+    public abstract void setMaxLatencyTier2(Long maxLatencyTier2);
+
+    @Override
+    @Property(PINNED_PROPERTY)
+    public abstract boolean isPinned();
+
+    @Override
+    @Property(PINNED_PROPERTY)
+    public abstract void setPinned(boolean pinned);
+
+    @Override
+    @Property(PRIORITY_PROPERTY)
+    public abstract Integer getPriority();
+
+    @Override
+    @Property(PRIORITY_PROPERTY)
+    public abstract void setPriority(Integer priority);
+
+    @Override
+    @Property(PATH_COMPUTATION_STRATEGY_PROPERTY)
+    @Convert(PathComputationStrategyConverter.class)
+    public abstract PathComputationStrategy getPathComputationStrategy();
+
+    @Override
+    @Property(PATH_COMPUTATION_STRATEGY_PROPERTY)
+    @Convert(PathComputationStrategyConverter.class)
+    public abstract void setPathComputationStrategy(PathComputationStrategy pathComputationStrategy);
+
+    @Override
+    @Property(FORWARD_PATH_ID_PROPERTY)
+    @Convert(PathIdConverter.class)
+    public abstract PathId getForwardPathId();
+
+    @Override
+    @Property(FORWARD_PATH_ID_PROPERTY)
+    @Convert(PathIdConverter.class)
+    public abstract void setForwardPathId(PathId forwardPathId);
+
+    @Override
+    @Property(REVERSE_PATH_ID_PROPERTY)
+    @Convert(PathIdConverter.class)
+    public abstract PathId getReversePathId();
+
+    @Override
+    @Property(REVERSE_PATH_ID_PROPERTY)
+    @Convert(PathIdConverter.class)
+    public abstract void setReversePathId(PathId reversePathId);
+
+    @Override
+    @Property(PROTECTED_FORWARD_PATH_ID_PROPERTY)
+    @Convert(PathIdConverter.class)
+    public abstract PathId getProtectedForwardPathId();
+
+    @Override
+    @Property(PROTECTED_FORWARD_PATH_ID_PROPERTY)
+    @Convert(PathIdConverter.class)
+    public abstract void setProtectedForwardPathId(PathId protectedForwardPathId);
+
+    @Override
+    @Property(PROTECTED_REVERSE_PATH_ID_PROPERTY)
+    @Convert(PathIdConverter.class)
+    public abstract PathId getProtectedReversePathId();
+
+    @Override
+    @Property(PROTECTED_REVERSE_PATH_ID_PROPERTY)
+    @Convert(PathIdConverter.class)
+    public abstract void setProtectedReversePathId(PathId protectedReversePathId);
+
+    @Override
+    public Optional<HaSubFlow> getSubFlow(String subFlowId) {
+        if (subFlows == null) {
+            // init the cache map with sub flows.
+            getSubFlows();
```

Comment:
no concurrency issues here? When two threads start initializing the map, then what?